packages/ui/src/models/visualization/flows/support/camel-component-schema.service.ts

```diff
@@ -464,7 +427,21 @@ export class CamelComponentSchemaService {
     const [pathUri, queryUri] = definition.uri?.split('?') ?? [undefined, undefined];
     if (queryUri) {
       definition.uri = pathUri;
-      Object.assign(definition.parameters, CamelUriHelper.getParametersFromQueryString(queryUri));
+
+      const validParametersFromQueryString = Object.entries(
```

Comment:
@shivamG640, just to confirm, here are we bringing only the properties that exists in the component? is that right?

Reply:
yes, That is the code from the old PR...

Reply:
I know we discussed to deprecate the code for parsing any URI parameters for routes with URI syntax but since we have that functionality now, user can add invalid parameters that way!

Reply:
I see, I think might hit an issue with the `toD` component, because there won't be any parameters I fear. Also, if someone has some properties from a ~newer~ different camel version, we would remove those parameters, non-existing for the current Camel version, but potentially existing for another version.

I think a more realistic example would be prefixed parameters, like `parameter.anotherParam`, those are valid and present in the URI field, but non-existing per se, the component definition only says parameters prefixed with `parameters`.

Here's [an example](https://github.com/apache/camel/blob/a484235733cdb9807dc7e57b63e428d8e108cbf8/catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/components/langchain4j-tools.json#L37), notice the `parameter.` prefix, it means the URI would look like:
```
uri: langchain4j-tools?parameter.location=string&parameter.hey=hello
```

If we follow this approach, wouldn't we ignore those `parameter.location` and `parameter.hey` parameters?

Reply:
I removed the components properties check